php/commands/src/CLI_Command.php

```diff
@@ -575,4 +575,96 @@ private function get_update_type_str( $assoc_args ) {
 		return $update_type;
 	}
 
+	/**
+	 * Detects if a command exists
+	 *
+	 * This commands checks if a command is registered with WP-CLI.
+	 * If the command is found then it returns with exit status 0.
+	 * If the command doesn't exist, then it will exit with status 1.
+	 *
+	 * ## OPTIONS
+	 * <command_name>...
+	 * : The command
+	 *
+	 * ## EXAMPLES
+	 *     wp cli has-command "wp site delete"
+	 *     wp cli has-command wp site delete
+	 *
+	 * @subcommand has-command
+	 *
+	 * @when before_wp_load
```

Comment:
Having `before_wp_load` is an issue as it won't detect commands added in plugins/themes. Actually, this is an issue for `cmd-dump` as well as the whole `CLI_Command` class is marked as `before_wp_load`. So I think the `before_wp_load` needs to be removed from the `CLI_Command` class and added to the individual commands, except for this command and `cmd-dump`, where `after_wp_load` should be used.

Reply:
> Actually, this is an issue for `cmd-dump` as well

Historically, `cmd-dump` has only been used for exporting bundled command definition to documentation, so it hasn't actually been an issue.

Reply:
Oh, ok - I'll open an issue re individual function `when`s not being honoured if there's a class `when` as it probably needs discussion.